var/spack/repos/builtin/packages/py-pytest/package.py

```diff
@@ -37,7 +38,8 @@ class PyPytest(PythonPackage):
     depends_on('py-setuptools@30.3:', when='@3.9.0:3.9.1', type=('build', 'run'))
     depends_on('py-setuptools', type=('build', 'run'))
     depends_on('py-setuptools-scm', when='@3.1:', type='build')
-    depends_on('py-py@1.5.0:', when='@3.3:', type=('build', 'run'))
+    depends_on('py-py@1.8.2:', when='@:6.0.0', type=('build', 'run'))
+    depends_on('py-py@1.5.0:', when='@3.3:6.0', type=('build', 'run'))
```

Comment:
```suggestion
    depends_on('py-py@1.8.2:', when='@6:', type=('build', 'run'))
    depends_on('py-py@1.5.0:', when='@3.3:5.999', type=('build', 'run'))
```
Is this what you're trying to do here?

Reply:
yes. Changes done.